recipes/bgfx/all/conanfile.py

```diff
@@ -48,119 +48,133 @@ def _genie_extra(self):
+                self.output.warning(f"{self.ref} does no checking for the current compiler. Assuming it works, please consider adding it.")
 
     def build_requirements(self):
         self.tool_requires("genie/1170")
         if not is_msvc(self) and self._settings_build.os == "Windows":
-            self.win_bash = True
-            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
-                self.tool_requires("msys2/cci.latest")
+            if self.settings.os == "Windows": # building for windows mingw
+                if "MINGW" not in os.environ:
+                    self.tool_requires("mingw-builds/13.2.0")
+            else: # cross-compiling for something else, probably android; get native make
+                self.tool_requires("make/[>=4.4.1]")
+        if self.settings.os == "Android" and "ANDROID_NDK_ROOT" not in os.environ:
+            self.tool_requires("android-ndk/[>=r26d]")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+        get(self, **self.conan_data["sources"][self.version]["bgfx"], strip_root=True,
                     destination=os.path.join(self.source_folder, self._bgfx_folder))
-        # bgfx's genie project, and the projects generated by it, expect bx and bimg source to be present on the same relative root as bimg's in order to build
-        # usins a pre-built bx and bimg instead would require significant changes to the genie project but may be worth looking into in the future
-        get(self, **self.dependencies["bx"].conan_data["sources"][self._bx_version[self.version]], strip_root=True,
+        # bgfx's genie project, and the projects generated by it, expect bx and bimg source to be present on the same relative root as bgfx's in order to build
+        # usins a pre-built bx and bimg instead would require significant changes to the genie project but may be worth looking into in the future, if upstream wants to go that route
+        get(self, **self.conan_data["sources"][self.version]["bx"], strip_root=True,
                     destination=os.path.join(self.source_folder, self._bx_folder))
-        get(self, **self.dependencies["bimg"].conan_data["sources"][self._bimg_version[self.version]], strip_root=True,
+        get(self, **self.conan_data["sources"][self.version]["bimg"], strip_root=True,
                     destination=os.path.join(self.source_folder, self._bimg_folder))
 
     def generate(self):
         vbe = VirtualBuildEnv(self)
         vbe.generate()
         if is_msvc(self):
-            tc = VCVars(self)
+            tc = MSBuildToolchain(self)
+            if not self.settings.get_safe("compiler.cppstd"):
+                tc.cppstd = "c++17"
             tc.generate()
         else:
             tc = AutotoolsToolchain(self)
+            if not self.settings.get_safe("compiler.cppstd"):
+                tc.cppstd = "-std=c++17"
```

Comment:
Filed as https://github.com/conan-io/conan/issues/16533

Reply:
@AbrilRBS What's the fate of this following the closure of #16533 ?